### PR TITLE
Tkasubosk42 signoff projectile path sensor

### PR DESCRIPTION
**Pull request by tkasubosk42**
requesting signoff for Projectile Path Sensor

### CODE REVIEW

**Review by JesseTNRoberts — COMMENTED** [review]
I gave general input regarding constraints to the team. This design has constraints that do not constrain this subsystem at all. Further they are imprecise (no units or value ranges). Both of these issues were discussed in general with the team yesterday. 

Constraints start with shall.

File: Documentation/Signoffs/ProjectilePathSensor.md [hunk 1]
```diff
@@ -0,0 +1,104 @@
...
+
+## Subsystem Function
+The projectile Path sensor subsystem must detect when a projectile is launched and determine which of the 15 possible paths the projectile will take. This is expected to be the first sensor data sent to the interceptor. This data provies the required yaw angle to properly intercept the incoming projectiles.
+## Constraints
+
+- Constrain 1: The Path Sensor Array shall be autonomous, and will require to no outside interference after initial startup.
+  - Reasoning: Do to a constumer requirement the interceptor and all external sensor post must be autonomous after intial startup.
+- Constraint 2: The Path Sensor Array shall be capable of detecting which of the possible 15 paths an approaching object is taking and relay this value to the interceptor.
+  - This constraint was developed to meet the requirement, to have a sensor that can determine when a projectile has begun to move towards the interceptor, and determine the path at which the projectile is heading towards to the interceptor.
+- Constraint 3: The Path Sensor Array shall that operates on battery power and is cappable of supplying power to the entire subsystem continously for 30 minutes.
```
Comment (JesseTNRoberts) [inline Documentation/Signoffs/ProjectilePathSensor.md]: What does this say/mean?

File: Documentation/Signoffs/ProjectilePathSensor.md [hunk 2]
```diff
@@ -0,0 +1,104 @@
...
+  - Due to a constraint given by Customer the sensor posts must have independent power from the interceptor.
+  - The Subsystem is required to be capable of supply power for 30 minutes to provide adaquate time to complete the competition.
+- Constraint 4: The Path Sensor Array shall implement the best remote battery power solution to limit the interceptor’s Envirmental Impact by designing a system that requires one set of batteries for the entirety of the project.
```
Comment (JesseTNRoberts) [inline Documentation/Signoffs/ProjectilePathSensor.md]: this is not a good constraint.

File: Documentation/Signoffs/ProjectilePathSensor.md [hunk 3]
```diff
@@ -0,0 +1,104 @@
...
+  - This constraint addresses the broader impacts of current enviromental impacts of disposing batteries.
+- Constaint 5: The Path Sensor Array shall comply with the ANSI Z136.1 Standard
```
Comment (JesseTNRoberts) [inline Documentation/Signoffs/ProjectilePathSensor.md]: It's not sufficient to say comply with... (some huge thing we haven't read most of)

Be specific. What are the constraints?

File: Documentation/Signoffs/ProjectilePathSensor.md [hunk 4]
```diff
@@ -0,0 +1,104 @@
...
+  - This constraint is required due to the use of laser sensors to determine when a projectile is launched. This Standards clasifies lasers as well as defines the required PPE while useing lasers.
+- Constraint 6: The Path Sensor Array shall implement a microcontroller that has 17 I/O pins, 15 Pin change interupts, 5 Volt pin.
```
Comment (JesseTNRoberts) [inline Documentation/Signoffs/ProjectilePathSensor.md]: So is this a constraint on the path sensor or the computing system or are they together?

**Review by JesseTNRoberts — COMMENTED** [review]
This design is good progress I believe. The major issue now is, you will need to have a 3d model of how all these are mounted to the posts as well.

**Comment by JTJones73** [comment]
"Do to a constumer requirement" -> "Due to a costumer"
"entire subsystem for 30 minutes. " Adam's subsystem states it must last for 1 hour we need to straighten this constraint out
"Customer" make lowercase
Show the esp being powered. You will probably need an exert about why you need an esp
Add some description text for the figures
"calculate the intensity of the lasers used the calculation required is I = P/A" Reword this sentence to be clearer
"I can estimate" Avoid personal pronouns
Several personal pronouns please reword
"Volts[1]." inconsistent spacing
"Do to battery" Due to
"atleast" at least
"Prcedural" typo
"thaThn .25 seconds" typo
"Current requirements" clarify if you are saying the amperage requirement is.. or currently the microcontroller is required to...
"6 sets of Terminal Blocks [21] 	NA 	1 	$13.99 	$13.99" No link but make it Terminal Blocks|NA|6|$perUnit|total

**Review by JesseTNRoberts — COMMENTED** [review]
Most of your constraints are questionable as to whether they are constraints or are design decisions. 

Also, fix the typographical errors and be sure that the overall look of the document is professional.

How precise does the placement of each laser have to be? What if the ball has a bit of movement coming down the track, will the laser still see it? How much movement will cause this system to not see the ball? What are the consequences?

This seems like you are capturing the wires in between the upper and lower pieces of wood. Is this ok? And how do you actually install that?

**Review by JesseTNRoberts — COMMENTED** [review]
Constraints weren't changed.

**Review by JesseTNRoberts — COMMENTED** [review]
Is the buck converter needed? (could the lasers be connected straight to the battery or could they be connected to the 5 volt rail of the micro?

Why are there two microcontrollers here?

How frequently does the microcontroller need to sample each of the sensors to ensure that a ball isn't missed?

This doesn't seem to include the external power supply you described.

**Review by JesseTNRoberts — COMMENTED** [review]
This all looks pretty good. A couple of issues

1. The constraints need to be cleaned up into a table with where they came from. 
2. One of the constraints comes from broader impacts regarding minimizing batteries. I don't see anything about that in the analysis.
3. The BOM has components that are irrelevant
4. You haven't found a rail for the sensors based on your BOM

**Review by JesseTNRoberts — COMMENTED** [review]
Very good work!

**Pull request merged**